### PR TITLE
2.4.1-RC2

### PR DESCRIPTION
**Pull request by Afayadas**
…eset module issue + fix use statement in 1.6

### CODE REVIEW

File: upgrade/Upgrade-2.4.1.php [hunk 1]
```diff
@@ -39,38 +39,94 @@ function upgrade_module_2_4_1($module)
...
+ * @throws PrestaShopDatabaseException
+ */
+function cleanStripeOrderState($name, $color)
+{
+    $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS(
+        'SELECT os.id_order_state FROM `' . _DB_PREFIX_ . 'order_state` os
+            INNER JOIN `' . _DB_PREFIX_ . 'order_state_lang` osl ON (os.id_order_state = osl.id_order_state)
+            WHERE osl.name LIKE "%' . $name . '%" AND os.color = "' . $color . '" GROUP BY os.id_order_state ORDER BY os.id_order_state DESC;');
+
+    $order_state_ids = array_column($result, 'id_order_state');
```
Comment (clotairer) [inline upgrade/Upgrade-2.4.1.php:116]: What's happend with array_column if $result eq false (no results) ?
Someone can update his color, in this case, the result will be empty.
You should return before if empty($result)

File: upgrade/Upgrade-2.4.1.php [hunk 2]
```diff
@@ -39,38 +39,94 @@ function upgrade_module_2_4_1($module)
...
+
+    $order_state_id = $order_state_ids[0];
+
+    if (count($order_state_ids) > 1) {
+        unset($order_state_ids[0]);
+
+        $order_state_to_clean = implode(',', $order_state_ids);
+
+        if (!Db::getInstance(_PS_USE_SQL_SLAVE_)->execute('UPDATE `' . _DB_PREFIX_ . 'order_history` SET id_order_state = ' . $order_state_id . ' WHERE id_order_state IN (' . $order_state_to_clean . ');')
```
Comment (clotairer) [inline upgrade/Upgrade-2.4.1.php:125]: You cannot do that. Probably other modules like ERP use PrestaShop state and history to manage synchronisation of order. you have no right to do that.

File: upgrade/Upgrade-2.4.1.php [hunk 3]
```diff
@@ -39,38 +39,94 @@ function upgrade_module_2_4_1($module)
...
+            || !Db::getInstance(_PS_USE_SQL_SLAVE_)->execute('DELETE FROM `' . _DB_PREFIX_ . 'order_state` WHERE id_order_state IN (' . $order_state_to_clean . ');')) {
```
Comment (clotairer) [inline upgrade/Upgrade-2.4.1.php:126]: You cannot do that. If you want to remove an state, please use the appropriate methode of the class that trigger hooks. States removed are just hidden to keep memory.

File: upgrade/Upgrade-2.4.1.php [hunk 4]
```diff
@@ -39,38 +39,94 @@ function upgrade_module_2_4_1($module)
...
+            return false;
+        }
+    }
+
+    return $order_state_id;
+}
```
Comment (clotairer) [inline upgrade/Upgrade-2.4.1.php:132]: This refactoring is to dangerous in a minor version.

File: upgrade/Upgrade-2.4.1.php [hunk 5]
```diff
@@ -39,38 +39,94 @@ function upgrade_module_2_4_1($module)
...
+                ['name' => 'oxxo', 'color' => '#C23416', 'config' => stripe_official::OXXO_WAITING],
+            ];
+
+            $order_state_to_affect = array_map(
+                'cleanStripeOrderState',
+                array_column($stripe_order_states, 'name'),
+                array_column($stripe_order_states, 'color')
+            );
+
+            for ($i = 0; $i < 5; $i++) {
```
Comment (clotairer) [inline upgrade/Upgrade-2.4.1.php:70]: foreach looks like easier to understand

File: upgrade/Upgrade-2.4.1.php [hunk 6]
```diff
@@ -39,38 +39,94 @@ function upgrade_module_2_4_1($module)
...
-            if (empty($result[0])) {
+        $result = Db::getInstance(_PS_USE_SQL_SLAVE_)->executeS(
+            'SELECT os.id_order_state FROM `' . _DB_PREFIX_ . 'order_state` os
+            INNER JOIN `' . _DB_PREFIX_ . 'order_state_lang` osl ON (os.id_order_state = osl.id_order_state)
+            WHERE osl.name LIKE "%stripe%" AND os.color = "#FFDD99" GROUP BY os.id_order_state;');
+
+        if (!empty($result)) {
+            $order_state_to_delete = array_column($result, 'id_order_state');
+
+            if (!Db::getInstance(_PS_USE_SQL_SLAVE_)->execute('DELETE FROM `' . _DB_PREFIX_ . 'order_state` WHERE id_order_state IN (' . implode(',', $order_state_to_delete) . ');')) {
```
Comment (clotairer) [inline upgrade/Upgrade-2.4.1.php:50]: use OrderState method to deleted to trigger hooks.

File: stripe_official.php [hunk 7]
```diff
@@ -511,51 +513,51 @@ public function install()
...
-            || !Configuration::updateValue(self::ENABLE_OXXO, 0, false, $shopGroupId, $shopId)) {
-                 return false;
-        }
-
-        if (!$this->installOrderState()) {
+            Stripe_officialClasslib\Extensions\ProcessLogger\ProcessLoggerHandler::closeLogger();
             return false;
         }
-
-        return true;
```
Comment (clotairer) [inline stripe_official.php:558]: Are you sure to remove return true at then end of install ?

**Review by clotairer — CHANGES_REQUESTED** [review]

File: stripe_official.php (hunk 7 above)
Comment (Afayadas) [reply stripe_official.php:558]: i have place the return true statement in the try block (line 541)

**Review by clotairer — APPROVED** [review]

**Pull request merged**